### ISSUE TITLE
Questions about training  process

### ISSUE THREAD

**Issue by iris0329** [Halmstad-University/SalsaNext#32]
Hi, 
Thanks for your generous opening source of this briliant project!

I have two questions for the project:
1. Does pretrained model use uncertainty or not during training? Is this pretrained model the one which could reproduce `59.5 point-wise mean-IoU` in the Table I of paper?

2.  When I train the model on my self, I found that there is a serious overfitting problem. 
The pictures below are my training loss and valid loss curve.  
I want to ask why this problem occurs? 
Do you also have this problem during training?

![image](https://user-images.githubusercontent.com/26380036/98063680-87a5a780-1e8b-11eb-868b-fe9719877471.png)
![image](https://user-images.githubusercontent.com/26380036/98063690-90967900-1e8b-11eb-9495-36406c4ead0f.png)

I am looking forward to your reply!

**Comment by TiagoCortinhal** [comment]
Hello @iris0329 

1. No the pretrained model is not trained with the uncertainty. The uncertainty is applied after training following [A General Framework for Uncertainty Estimation in Deep Learning](https://arxiv.org/abs/1907.06890)

2. Could you give me more information about your training? I never encountered this type of issues, no.

**Comment by iris0329** [comment]
Thanks for your prompt reply!

1. So that means, if I apply uncertainty on this pretrained model, I could get `59.5 point-wise mean-IoU` in the Table I of paper?

2. The batch size is set to 32 and I used 8 blocks of TITAN X (Pascal) GPU. All the other Settings are the default Settings.
It looks very confusing
![image](https://user-images.githubusercontent.com/26380036/98093885-7ed0c800-1ec3-11eb-8dc1-383e7a12ffa6.png)

I retrained it, and it can be seen that there is a tendency of overfitting at **EPOCH 55.**
Thanks.

**Comment by TiagoCortinhal** [comment]
1. Yes, which was evaluated on the test set.

2. Does that tendency continues if it trains after 55 epochs? 


Best,

**Comment by iris0329** [comment]
1. Thank you 
2. The answer is yes. Now It's nearly 100 epoch, you could see from the validation and training loss figure that overfitting is becoming serious

![image](https://user-images.githubusercontent.com/26380036/98210343-6328f880-1f7b-11eb-8f96-9382a23c7fad.png)

![image](https://user-images.githubusercontent.com/26380036/98210546-b0a56580-1f7b-11eb-8ebf-d2e0f807132d.png)

**Comment by TiagoCortinhal** [comment]
Ok I will check this as soon as possible!

**Comment by TiagoCortinhal** [comment]
Hello @iris0329! Sorry for the delay on this question. I had to go back to my logs to check how the submission training (and the other experiments went) to check our behaviour. 

I mislead you earlier and we had a similar issue with an apparent overfitting. Also our baseline comparison (rangenet++) also shows this apparent sign when we trained it from scratch. 

Again, sorry for the delay and misleading answers before, I didn't have in my mind a clear idea of our val_loss plots!

**Comment by iris0329** [comment]
Sorry, I thought about this problem before, but forgot to update it here.

I think it is a problem caused by Cross-Entropy Loss.

Think about the Matrix_A before argmax function, its size is (h, w, class_num). After applying the argmax function on this Matrix_A, we got a new Matrixd_B and its size is (h, w, 1).

The increment IoU means that the Matrix_B becomes more accurate. However, the Cross-Entropy Loss is computed based on Matrix_A.

For example, 
> There are two pixels and ground truth is [2, 0]. At the beginning, Matrix_A is [[0.1, 0.2, 0.7], [0.1, 0.2, 0.7]], after applying the argmax function, we got the prediction [2, 2], 50% accuracy rate.

> With the training process continuing,  the Matrix_A becomes to [[0.2, 0.3, 0.5], [0.5, 0.3, 0.2]] after applying the argmax function, the prediction is [2, 0], the loss will increase but the accuracy is higher.
 
So it seems, as the training progressed, there was a tendency for the predictions to average out across all categories. But I still don't know how to solve it. If anyone has a suggestion about it, I would be very grateful.
